src/SFA.DAS.Charities.Data/Repositories/CharityImportRepository.cs

```diff
@@ -27,8 +30,20 @@ public async Task ClearStagingData()
         {
             using var tx = await _charitiesDataContext.Database.BeginTransactionAsync();
 
-            await _charitiesDataContext.Database.ExecuteSqlInterpolatedAsync($"delete from CharityTrusteeStaging");
-            await _charitiesDataContext.Database.ExecuteSqlInterpolatedAsync($"delete from CharityStaging");
+            await _charitiesDataContext.Database.ExecuteSqlInterpolatedAsync($"TRUNCATE TABLE CharityTrusteeStaging");
+            await _charitiesDataContext.Database.ExecuteSqlInterpolatedAsync($"TRUNCATE TABLE CharityStaging");
+
+            await tx.CommitAsync();
+        }
+
+        public async Task LoadDataFromStagingInToLive()
+        {
+            using var tx = await _charitiesDataContext.Database.BeginTransactionAsync();
+
+            await _charitiesDataContext.Database.ExecuteSqlInterpolatedAsync($"TRUNCATE TABLE CharityTrustee");
+            await _charitiesDataContext.Database.ExecuteSqlInterpolatedAsync($"TRUNCATE TABLE Charity");
+            await _charitiesDataContext.Database.ExecuteSqlRawAsync("INSERT INTO Charity SELECT * FROM CharityStaging");
```

Comment:
If you smoke tested and this worked, no problem with this.  I would expect this not to work cos inserting over the primary key, but if it works, that;s fine.

Reply:
You can insert into primary key as long as it is not an identity column.